### PR TITLE
fix(lambda-promtail): Remove s3 bucket ownership verification

### LINKED ISSUE #15382: lambda-promtail: s3 bucket ownership verification not working
The lambda-promtail implementation uses s3 bucket ownership verification when copying logs from s3 buckets, but it is incorrectly using the s3 bucket ownerIdentity principal id as aws account id which produces the following error:

api error InvalidBucketOwnerAWSAccountID: The value of the expected bucket owner parameter must be an AWS Account ID...

S3 bucket ownership verification seems to be only possible by verifying the aws account id: 
https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucket-owner-condition.html
However the S3 event notification message which the current implementation uses to get the AWS S3 account id doesn't actually contain it: 
https://docs.aws.amazon.com/AmazonS3/latest/userguide/notification-content-structure.html
https://docs.aws.amazon.com/lambda/latest/dg/with-s3.html

It seems that there is no way to get the AWS account id from the S3 event notification message as it is considered sensitive information.
Suggestion is to remove the s3 bucket ownership verification, otherwise the implementation will always produce an error.

### PR DESCRIPTION
**Pull request by milenpankov-bosch**
**What this PR does / why we need it**: Removes the s3 bucket ownership verification as it is not using an AWS account id for verification.

**Which issue(s) this PR fixes**:
Fixes #15382

### CODE REVIEW

File: tools/lambda-promtail/lambda-promtail/s3.go [hunk 1]
```diff
@@ -252,7 +252,6 @@ func getLabels(record events.S3EventRecord) (map[string]string, error) {
 
 	labels["key"] = record.S3.Object.Key
 	labels["bucket"] = record.S3.Bucket.Name
-	labels["bucket_owner"] = record.S3.Bucket.OwnerIdentity.PrincipalID
```
Comment (chaudum) [inline tools/lambda-promtail/lambda-promtail/s3.go:255]: Should we keep the `bucket_owner` label and only remove the verification? I'm not sure if anyone relies on the label being present.

**Review by chaudum — COMMENTED** [review]
Otherwise LGTM

File: tools/lambda-promtail/lambda-promtail/s3.go [hunk 2]
```diff
@@ -292,10 +291,9 @@ func processS3Event(ctx context.Context, ev *events.S3Event, pc Client, log *log
 			&s3.GetObjectInput{
 				Bucket:              aws.String(labels["bucket"]),
 				Key:                 aws.String(labels["key"]),
-				ExpectedBucketOwner: aws.String(labels["bucket_owner"]),
 			})
 		if err != nil {
-			return fmt.Errorf("failed to get object %s from bucket %s on account %s, %s", labels["key"], labels["bucket"], labels["bucket_owner"], err)
+			return fmt.Errorf("failed to get object %s from bucket %s on account %s, %s", labels["key"], labels["bucket"], err)
```
Comment (chaudum) [inline tools/lambda-promtail/lambda-promtail/s3.go:296]:
```suggestion
			return fmt.Errorf("failed to get object %s from bucket %s, %s", labels["key"], labels["bucket"], err)
```

File: tools/lambda-promtail/lambda-promtail/s3.go (hunk 1 above)
Comment (milenpankov-bosch) [reply tools/lambda-promtail/lambda-promtail/s3.go:255]: As far as I can see it is not used anywhere else, so I guess it is not needed.

Comment (chaudum) [reply tools/lambda-promtail/lambda-promtail/s3.go:255]: The `labels` are part part of the payload sent to Loki. So, someone might be missing this label after your change.

Comment (milenpankov-bosch) [reply tools/lambda-promtail/lambda-promtail/s3.go]: I have reverted the label.

**Review by chaudum — APPROVED** [review]

**Pull request merged**